### DISCUSSION TITLE
Disable/Enable Short URLs?

### DISCUSSION THREAD

**Discussion by YouveGotMeowxy** [YOURLS/YOURLS#3840] [Plugins]
I often use short urls to point to my affiliate links in a blog, and I swear these affiliate companies hate us publishers because they're constantly change these up, silently, and we end up with dead affiliate links for ages w/o knowing it.

When I find out I'd like to be able to simply disable that sort URL until I find another replacement affiliate for it, and not keep sending the previous affiliate free traffic.

It seems this is best for a plugin but I wouldn't even know where to begin to create this myself, as my coding skills are pretty basic.

**Comment by dgw** [comment]
This came up in #3851. Directing the topic here, as it's unlikely to be a core feature.

**Reply by YouveGotMeowxy** [reply]
Out of curiosity, why not make it a core feature? Enabling/disabling these short links seems pretty fundamental to an app that shorten links?

**Reply by dgw** [reply]
It can't be that fundamental a feature if market leader Bitly doesn't have it… All I can find for them is "hide" (from your own dashboard) and "delete" (irreversibly) actions.

I've seen it requested twice in 5+ years of active participation in this project. Twice, among unknown hundreds (thousands?) of users.

A plugin should do this.

**Reply by YouveGotMeowxy** [reply]
Are TinyURL and OWL.LY market leader enough?

![image](https://github.com/user-attachments/assets/c34dc4af-bec3-4a96-a33d-6db9794b9142)

One reason I (and probably others) don't use bitly and use yourls instead is lack of some features. Unless you are merely aiming to copy bitly feature for feature. *shrug*

**Reply by dgw** [reply]
"Generative AI is experimental." You're not even the first person _today_ to just send me a screenshot of Google's half-baked AI answers and treat it as truth without checking the primary source yourself.

The tinyurl help center has [zero results for "disable"](https://helpdesk.tinyurl.com/search?q=disable); Google seems to be [hallucinating](https://en.wikipedia.org/wiki/Hallucination_(artificial_intelligence)) that feature.

For Ow.ly, Google's misinterpreted the task request, and instead described how to turn on and off using Ow.ly to shorten links in the posts you send through HootSuite (instead of the full link). It's like if Twitter offered a toggle for t.co.

In nearly 20 years, I personally have _never_ seen a URL shortener with this "enable"/"disable" feature you seem to have in mind. Create, edit destination, edit keyword, and delete are the basic operations, and YOURLS has all of those. AFAICT, those are also the operations tinyurl and bitly support.

YOURLS also has a lot of existing plugins, one of which could be used to do this with a bit of creativity: https://github.com/district09/yourls_plugin_http-status-per-link — Setting 302 or 307 status for the links you plan to "disable" can be used in conjunction with a landing page explaining that the link is unavailable right now. (Append a `#unique_fragment` if your YOURLS settings disallow duplicates.)

Switching the HTTP status code would be important anyway for links you expect to change, because the default 301 redirect is aggressively cached by every major browser (and probably the little ones too).

----

My opinion isn't the final word on this. I'm not that involved with development. But I am trying to give you a realistic expectation based on years of collaborating with the rest of the core team. We're focused on keeping the core simple, and this feature really sounds too niche to me for inclusion in core at this time. (@YOURLS/core do chime in if you disagree! I know y'all have busy lives these days, but I hope you read these when you get the spare time 😁)

**Reply by LeoColomb** [reply]
I cannot give a better answer than what @dgw said. 
Would bring technical challenges, no high demand of the feature, and doable with a plugin.

**Reply by YouveGotMeowxy** [reply]
> Would bring technical challenges, doable with a plugin.

I've never looked into what is required to create a plugin, what is less challenging about doing it via a "hacked-in" plugin that is more technically challenging to add to the core code itself?

Mind you, I personally don't care whether it's in core or plugin form, I'm just curious about the thought process involved.
